lib/Value/Matrix.pm

```diff
@@ -3,6 +3,109 @@
 #  Implements the Matrix class.
 #
 #    @@@ Still needs lots of work @@@
+
+=head1 Value::Matrix class
+
+
+References:
+
+MathObject Matrix methods: L<http://webwork.maa.org/wiki/Matrix_(MathObject_Class)>
+MathObject Contexts: L<http://webwork.maa.org/wiki/Common_Contexts>
+CPAN RealMatrix docs: L<http://search.cpan.org/~leto/Math-MatrixReal-2.09/lib/Math/MatrixReal.pm>
+
+Allowing Matrices in Fractions: 
+L<http://webwork.maa.org/moodle/mod/forum/discuss.php?d=2978>	
+
+     Context()->parens->set("[" => {formMatrix => 1});
+
+Files interacting with Matrices:
+
+L<lib/MatrixReal1.pm> L<http://webwork.maa.org/pod/pg_TRUNK/macros/MatrixReduce.pl.html>
+L<lib/Matrix.pm> 
+
+L<macros/MatrixCheckers.pl> -- checking whether vectors form a basis
+L<macros/MatrixReduce.pl>  -- tools for  row reduction via elementary matrices
+L<macros/MatrixUnits.pl>   -- Generates unimodular matrices with real entries
+L<macros/PGmatrixmacros.pl>
+L<macros/PGmorematrixmacros.pl>
+L<macros/PGnumericalmacros.pl>
+L<macros/tableau.pl>
+L<macros/quickMatrixEntry.pl>
+L<macros/LinearProgramming.pl>	
+Contexts
+
+	Matrix -- allows students to enter [[3,4],[3,6]] 
+	       -- formMatrix =>1 also allows this?
+	Complex-Matrix -- allows complex entries
+
+Creation methods
+
+		 $M1 = Matrix([1,2],[3,4]);
+   		 $M2 = Matrix([5,6],[7,8]);
+    	  $v = Vector(9,10);
+    	  $w = ColumnVector(9,10); # differs in how it is printed
+  
+Commands added in Value::matrix
+
+	Conversion
+		$matrix->values produces [[3,4,5],[1,3,4]] recursive array references of numbers (not MathObjects)
+		$matrix->wwMatrix   produces CPAN MatrixReal1 matrix, used for computation subroutines
+
+	Information
+		$matrix->dimension:  ARRAY
+
+	Access values
+
+		row : MathObjectMatrix
+		column : MathObjectMatrix
+		element : Real or Complex value
+
+	Assign values
+
+		these need to be added:
+		
+see C<change_matrix_entry()> in MatrixReduce and L<http://webwork.maa.org/moodle/mod/forum/discuss.php?d=2970>
+	
+	Advanced
+		$matrix->data:  ARRAY reference (internal data) of MathObjects (Real,Complex, Fractions)
+		                stored at each location.
+
+
+Passthrough methods covering subroutines in Matrix.pm which overrides or 
+augment CPAN's MatrixReal1.pm.  Matrix is a specialized subclass of MatrixReal1.pm
+
+The actual calculations are done in Matrix.pm
+
+	 trace 
+	 proj 
+	 proj_coeff 
+	 L 
+	 R 
+	 PL 
+	 PR
+
+Passthrough methods covering subroutines in MatrixReal1.pm 
+(this has been modified to handle complex numbers)
+The actual calculations are done in MatrixReal1.pm subroutines
+
+	condition
+	det 
+	inverse 
+	is_symmetric
+	decompose_LR 
+	dim 
+	norm_one 
+	norm_max 
+	kleene 
+	normalize 
+	solve_LR (also solve())
+	order_LR (also order()
```

Comment:
Missing a close paren.  :-)

Reply:
OK.  I fixed these and added some clarifying remarks.  For a more complete description one still needs to consult the MatrixReal1.pm documentation.